### PR TITLE
Bsc#1083455

### PR DESCRIPTION
**Pull request by charlesa**
This request contains fixes for two bugs.
bsc#1083455: This bug has a fix in libvirt, yast2-vm, and eventually yast2-network.
bsc#1086487: Spec file change to allow building and shipping yast2-vm on ppc64le.
If I can get a review done soon enough this can make it into SLE15 RC3.

### CODE REVIEW

File: src/modules/RelocationServer.rb [hunk 1]
```diff
@@ -261,7 +261,7 @@ def ReadLibvirtServices
       end
 
       ports = SuSEFirewallServices.GetNeededTCPPorts(
```
Comment (teclator) [inline src/modules/RelocationServer.rb]: SuSEFirewall should be replaced by firewalld. I already checked that the service definition is already provided by the libvirt package, the only question if there is some firewalld default service that already provides that definition and if not if the definition should be added in upstream or to our own package.

File: src/modules/VirtConfig.rb [hunk 2]
```diff
@@ -610,17 +570,6 @@ def LXCDialog
       Builtins.y2milestone("Start virtlogd.socket: %1", cmd)
       SCR.Execute(path(".target.bash"), cmd)
 
-      # Firewall stage - modify the firewall setting, add the xen bridge to FW_FORWARD_ALWAYS_INOUT_DEV
-      # Progress::NextStage();
-
-      # Configure firewall to allow xenbr+
```
Comment (teclator) [inline src/modules/VirtConfig.rb]: The code is commented (so not used) but also it is not supported at all by firewalld, so removed completely the setup.

File: src/modules/RelocationServer.rb [hunk 3]
```diff
@@ -245,24 +257,28 @@ def ReadLibvirtServices
...
-      if Service.Status("libvirtd") == 0
+      if Service.active?("libvirtd") == 0
         @libvirtd_is_running = true
         Builtins.y2milestone("libvirtd is running")
       else
         @libvirtd_is_running = false
         Builtins.y2milestone("libvirtd is not running")
       end
-      if Service.Status("sshd") == 0
+      if Service.active?("sshd") == 0
```
Comment (teclator) [inline src/modules/RelocationServer.rb]: active? method returns a boolean, so the comparison with 0 is not valid anymore.

File: src/modules/RelocationServer.rb [hunk 4]
```diff
@@ -245,24 +257,28 @@ def ReadLibvirtServices
       @libvirtd_enabled = Service.Enabled("libvirtd")
       @sshd_enabled = Service.Enabled("sshd")
 
-      if Service.Status("libvirtd") == 0
+      if Service.active?("libvirtd") == 0
```
Comment (teclator) [inline src/modules/RelocationServer.rb]: active? method returns a boolean, so the comparison with 0 is not valid anymore.

File: src/modules/RelocationServer.rb [hunk 5]
```diff
@@ -159,6 +166,9 @@ def ReadXendSettings
       true
     end
 
+    FWD_XEND_SERVICE = "xend-relocation-server".freeze
```
Comment (teclator) [inline src/modules/RelocationServer.rb]: It is not ready, as the xend service is not defined. At least I have not seen it.

**Comment by teclator** [comment]
@charlesa I have rebased the PR from master for resolving the conflicts once @mchf increased the version and forced a push.

File: package/yast2-vm.changes [hunk 6]
```diff
@@ -1,3 +1,18 @@
+-------------------------------------------------------------------
+Wed Apr  4 11:28:38 MDT 2018 - carnold@suse.com
+
+- bsc#1086487 - YaST2 segfault when virt-manager is installed but
+  not yast2-vm
+- 3.2.7
```
[the remark this answers is not in the document]
Comment (teclator) [reply package/yast2-vm.changes]: BTW, it is fine so. So just ignore the previous comment

File: src/modules/RelocationServer.rb (hunk 5 above)
Comment (teclator) [reply src/modules/RelocationServer.rb]: But it probably could be merged as in the worst case just firewalld will not be touched at all as the service until the service is defined.

Comment (charlesa) [reply src/modules/RelocationServer.rb]: All things xend need to be removed. Xend is only supported on SLE11 and older. This and some newer bugs entered need fixing. I will probably do a new pull request with all that is about to change.

**Comment by charlesa** [comment]
Created new pull request with significant additional changes. Dropping this one

**Pull request closed without merging**